File: 백준/Silver/1260. DFS와 BFS/DFS와 BFS.java
import java.io.*;
import java.util.*;

public class Main {
    static BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
    static BufferedWriter bw = new BufferedWriter(new OutputStreamWriter(System.out));
    static StringTokenizer st;
    static int n, m, v, a, b;
    static int[][] map;
    static boolean[] visited;
    static Queue<Integer> queue;


    static void read() throws IOException {
        st = new StringTokenizer(br.readLine());
        n = Integer.parseInt(st.nextToken());
        m = Integer.parseInt(st.nextToken());
        v = Integer.parseInt(st.nextToken());

        map = new int[n + 1][n + 1];
        for (int i = 0; i < m; i++) {
            st = new StringTokenizer(br.readLine());
            a = Integer.parseInt(st.nextToken());
            b = Integer.parseInt(st.nextToken());

            map[a][b] = map[b][a] = 1;
        }

        visited = new boolean[n + 1];
        dfs(v);
        bw.write("\n");
        visited = new boolean[n + 1];
        queue = new LinkedList<>();
        bfs(v);
    }

    static void dfs(int v) throws IOException {
        visited[v] = true;
        bw.write(v + " ");
        for (int i = 1; i <= n; i++) {
            if (map[v][i] == 1 && !visited[i]) dfs(i);
        }
    }

    static void bfs(int v) throws IOException {
        queue.add(v);
        visited[v] = true;

        while (!queue.isEmpty()) {
            v = queue.poll();
            bw.write(v + " ");

            for (int i = 1; i <= n; i++) {
                if (map[v][i] == 1 && !visited[i]) {
                    queue.add(i);
                    visited[i] = true;
                }
            }
        }
    }

    public static void main(String[] args) throws IOException {
        read();
        bw.flush();
        bw.close();
    }
}

File: 백준/Silver/1260. DFS와 BFS/DFS와 BFS.py
import sys

from collections import deque


def dfs(graph, v, visited):
    visited[v] = True
    print(v, end=' ')
    for i in range(1, len(graph)):
        if not visited[i] and graph[v][i] == 1:
            dfs(graph, i, visited)


def bfs(graph, v):
    queue = deque([v])
    visited = [False] * len(graph)
    visited[v] = True

    while queue:
        v = queue.popleft()
        print(v, end=' ')
        for i in range(1, len(graph)):
            if not visited[i] and graph[v][i] == 1:
                queue.append(i)
                visited[i] = True


input = sys.stdin.readline
n, m, v = map(int, input().split())
graph = [[0] * (n + 1) for _ in range(n + 1)]

for i in range(m):
    s, e = map(int, input().split())
    graph[s][e] = 1
    graph[e][s] = 1

visited = [False] * (n + 1)
dfs(graph, v, visited)
print()
bfs(graph, v)